test/relayserver/TransactionManager.test.ts

```diff
@@ -0,0 +1,323 @@
+import { PrefixedHexString, Transaction } from 'ethereumjs-tx'
+import { toBN } from 'web3-utils'
+import Mutex from 'async-mutex/lib/Mutex'
+import * as ethUtils from 'ethereumjs-util'
+
+import { deployHub, evmMineMany, revert, snapshot } from '../TestUtils'
+import { RelayServer } from '../../src/relayserver/RelayServer'
+import {
+  assertTransactionRelayed,
+  bringUpNewRelay,
+  getTimestampTempWorkdir, NewRelayParams,
+  PrepareRelayRequestOption,
+  relayTransaction,
+  RelayTransactionParams,
+  ServerTestConstants
+} from './ServerTestUtils'
+import { Address } from '../../src/relayclient/types/Aliases'
+import Web3 from 'web3'
+import { HttpProvider } from 'web3-core'
+import { RelayClient } from '../../src/relayclient/RelayClient'
+import { TestPaymasterEverythingAcceptedInstance } from '../../types/truffle-contracts'
+import { GsnRequestType } from '../../src/common/EIP712/TypedRequestData'
+import { GSNConfig } from '../../src/relayclient/GSNConfigurator'
+
+const TestPaymasterEverythingAccepted = artifacts.require('TestPaymasterEverythingAccepted')
+const TestRecipient = artifacts.require('TestRecipient')
+const StakeManager = artifacts.require('StakeManager')
+const Penalizer = artifacts.require('Penalizer')
+const Forwarder = artifacts.require('Forwarder')
+
+contract('TransactionManager', function (accounts) {
```

Comment:
If we're at it, this should also cover `sendTransaction()` missing tests... should open jira issue for that at least